liota/utilities/si_unit.py

```diff
@@ -0,0 +1,324 @@
+    else: # Prefixed unit, or non-SI unit
+        try:
+            pf = _get_prefix(tn[0])
+            un = tn[1]
+            # if re.compile("\s").search(str(un)) is not None:
+            if re.compile("\d").search(str(un)) is not None:
+                raise UnsupportedUnitError(unit)
+        except KeyError:
+            raise UnsupportedUnitError(unit)
+
+    # Return prefix and unit strings
+    return (pf, _get_unit_name(un))
+
+#---------------------------------------------------------------------------
+# These are units defined in tables from standard documents.
+# Table 1 include seven base units.
+# Table 2-4 include examples of derived units.
+# We want to make sure all units defined in these tables are parsed as they 
+# are specified in standard documents.
+# For derived units not in these tables, we deliver out best effort.
+
+units_table_1 = lambda ureg: [
+        ureg.m,     ureg.kg,    ureg.s,     ureg.A,     ureg.K,
+        ureg.mol,   ureg.cd
+    ]
+units_table_2 = lambda ureg: [
+        ureg.m ** 2,            ureg.m ** 3,
+        ureg.m / ureg.s,        ureg.m / ureg.s ** 2,
+        ureg.m ** -1,
+        ureg.kg / ureg.m ** 3,  ureg.kg / ureg.m ** 2,
+        ureg.m ** 3 / ureg.kg,
+        ureg.A / ureg.m ** 2,   ureg.A / ureg.m,
+        ureg.mol / ureg.m ** 3, ureg.kg / ureg.m ** 3,
+        ureg.cd / ureg.m ** 2,
+        ureg.dimensionless,     ureg.dimensionless
+    ]
+units_table_3 = lambda ureg: [
+        ureg.rad,   ureg.sr,    ureg.Hz,    ureg.N,     ureg.Pa,
+        ureg.J,     ureg.W,     ureg.C,     ureg.V,     ureg.F,
+        ureg.ohm,   ureg.S,     ureg.Wb,    ureg.T,     ureg.H,
+        ureg.degC,  ureg.lm,    ureg.lx,    ureg.Bq,    ureg.Gy,
+        ureg.Sv,
+        ureg.mol / ureg.s # katal (kat) is not supported by pint at this time
+    ]
+units_table_4 = lambda ureg: [
+        ureg.Pa * ureg.s,
+        ureg.N * ureg.m,        ureg.N / ureg.m,
+        ureg.rad / ureg.s,      ureg.rad / ureg.s ** 2,
+        ureg.W / ureg.m ** 2,
+        ureg.J / ureg.K,        ureg.J / (ureg.kg * ureg.K),
+        ureg.J / ureg.kg,       ureg.W / (ureg.m * ureg.K),
+        ureg.J / ureg.m ** 3,
+        ureg.V / ureg.m,        ureg.C / ureg.m ** 3,
+        ureg.C / ureg.m ** 2,   ureg.C / ureg.m ** 2,
+        ureg.F / ureg.m,        ureg.H / ureg.m,
+        ureg.J / ureg.mol,      ureg.J / (ureg.mol * ureg.K),
+        ureg.C / ureg.kg,       ureg.Gy / ureg.s,
+        ureg.W / ureg.sr,       ureg.W / (ureg.m ** 2 * ureg.sr),
+        (ureg.mol / ureg.s) / ureg.m ** 3
+    ]
+unit_tables = lambda ureg: [
+        units_table_1(ureg),
+        units_table_2(ureg),
+        units_table_3(ureg),
+        units_table_4(ureg)
+    ]
+
+
+# Testing code
+def main():
+    ureg = pint.UnitRegistry()
+    print_split = "-" * 76
+
+    # Base units and examples of derived units defined in SI standard documents
+    def parse_unit_with_color(un):
+        pf = "\033[1;36mnull\033[0m"
```

Comment:
These values are not intuitive, can you please add  a comment what do they mean?